### PR TITLE
applications: nrf5340_audio: use gpio-dt-spec

### PR DESCRIPTION
**Pull request by gmarull**
Use gpio-dt-spec facilities to simplify code in the module. Note that
module should be refactored to make better use of Devicetree (it relies
on device labels to make some decisions, something that needs to be
fixed).

### CODE REVIEW

File: applications/nrf5340_audio/src/modules/led.c [hunk 1]
```diff
@@ -31,19 +28,12 @@ LOG_MODULE_REGISTER(led, CONFIG_LOG_LED_LEVEL);
...
 
 /* The following arrays are populated compile time from the .dts*/
-const char *led_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), LABEL_AND_COMMA) };
-
-const int led_pins[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_AND_COMMA) };
-
-const char *led_gpio_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_LABEL_AND_COMMMA) };
-
-const int led_gpio_flags[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_FLAGS_AND_COMMMA) };
+const struct device leds[] = {
```
Comment (anangl) [inline applications/nrf5340_audio/src/modules/led.c:34]:
```suggestion
static const struct gpio_dt_spec leds[] = {
```

File: applications/nrf5340_audio/src/modules/led.c [hunk 2]
```diff
@@ -81,64 +65,38 @@ static uint8_t leds_num;
...
-	led_units[led_unit].type.mono.pol = (bool)led_gpio_flags[led];
-
-	if (led_units[led_unit].type.mono.dev == NULL) {
-		LOG_ERR("Could not get binding for %s", led_gpio_labels[led]);
+	if (!device_is_ready(leds[led].port)) {
+		LOG_ERR("LED GPIO controller not ready");
 		return -ENODEV;
 	}
 
+	led_units[led_unit].type.color = &leds[led];
```
[the remark this answers is not in the document]
Comment (gmarull) [reply applications/nrf5340_audio/src/modules/led.c]: oops fixed

File: applications/nrf5340_audio/src/modules/led.c [hunk 3]
```diff
@@ -81,64 +65,38 @@ static uint8_t leds_num;
...
-	led_units[led_unit].type.color[led_color].dev = device_get_binding(led_gpio_labels[led]);
-
-	if (led_units[led_unit].type.color[led_color].dev == NULL) {
-		LOG_ERR("Could not get binding for %s", led_gpio_labels[led]);
+	if (!device_is_ready(leds[led].port)) {
+		LOG_ERR("LED GPIO controller not ready");
 		return -ENODEV;
 	}
 
+	led_units[led_unit].type.color = &leds[led];
```
[the remark this answers is not in the document]
Comment (gmarull) [reply applications/nrf5340_audio/src/modules/led.c]: fixed

File: applications/nrf5340_audio/src/modules/led.c [hunk 4]
```diff
@@ -31,19 +28,12 @@ LOG_MODULE_REGISTER(led, CONFIG_LOG_LED_LEVEL);
...
-const char *led_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), LABEL_AND_COMMA) };
-
-const int led_pins[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_AND_COMMA) };
-
-const char *led_gpio_labels[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_LABEL_AND_COMMMA) };
-
-const int led_gpio_flags[] = { DT_FOREACH_CHILD(DT_PATH(leds), GPIO_FLAGS_AND_COMMMA) };
+static const struct device leds[] = {
+	DT_FOREACH_CHILD(DT_PATH(leds), GPIO_DT_SPEC_GET_AND_COMMA)
+};
```
Comment (gmarull) [inline applications/nrf5340_audio/src/modules/led.c:36]: FYI: raised this proposal upstream: https://github.com/zephyrproject-rtos/zephyr/pull/47342

File: applications/nrf5340_audio/src/modules/led.c [hunk 5]
```diff
@@ -31,19 +28,12 @@ LOG_MODULE_REGISTER(led, CONFIG_LOG_LED_LEVEL);
 #define CORE_APP_STR "CORE_APP"
 #define CORE_NET_STR "CORE_NET"
 
-#define LABEL_AND_COMMA(node_id) DT_LABEL(node_id),
-#define GPIO_AND_COMMA(node_id) DT_GPIO_PIN(node_id, gpios),
-#define GPIO_LABEL_AND_COMMMA(node_id) DT_GPIO_LABEL(node_id, gpios),
-#define GPIO_FLAGS_AND_COMMMA(node_id) DT_GPIO_FLAGS(node_id, gpios),
+#define GPIO_DT_SPEC_GET_AND_COMMA(node_id) GPIO_DT_SPEC_GET(node_id),
```
Comment (alexsven) [inline applications/nrf5340_audio/src/modules/led.c:31]:
```suggestion
#define GPIO_DT_SPEC_GET_AND_COMMA(node_id) GPIO_DT_SPEC_GET(node_id, gpios),
```

I don't see how the label is used anymore. The label was used to get what core the gpio should be assigned to, as well as telling if it's a mono or color LED

Comment (alexsven) [reply applications/nrf5340_audio/src/modules/led.c:31]: When I try to build this (after adding gpios as stated above) I get:
<err> led: No match for led unit. The dts is likely not properly formatted
<err> main: ERR_CHK Err_code: [-6] @ line: 147

Comment (gmarull) [reply applications/nrf5340_audio/src/modules/led.c]: added back the `led_labels`. I realized that code is parsing LED labels at runtime to determine things like its unit or core. We need to fix that by describing things properly in DT, labels are not meant to store information but just to give a name (useful e.g. to debug).

I'll open an issue to fix that in a follow-up PR.

**Review by alexsven — APPROVED** [review]
Verified that the latest commit works in our project

**Pull request merged**